### PR TITLE
Add a basic c sharp dot net blueprint

### PR DESCRIPTION
**Pull request by JoeDM09**

### CODE REVIEW

**Comment by JoeDM09** [comment]
@jackrk @kahouieong 
Just wanted to quickly make everyone aware that since initiating this pull request I have re-write the function to make this blueprint consistent with the Node blueprint and add better support for building IAM policies and conditions.

File: blueprints/dotnet/src/APIGatewayAuthorizerHandler.Tests/IntegrationTests.cs [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+using Amazon.Lambda.TestUtilities;
+using APIGatewayAuthorizerHandler.Model;
+using Newtonsoft.Json;
+using Xunit;
+
+namespace APIGatewayAuthorizerHandler.Tests
+{
+    public class IntegrationTests
+    {
+        [Fact]
```
Comment (JoeDM09) [inline blueprints/dotnet/src/APIGatewayAuthorizerHandler.Tests/IntegrationTests.cs]: This is just a really basic integration test to help attach the debugger, I'd like to add proper unit testing later, but for now this allows for a smoke test and lets the developer manually test at least a little bit.

File: blueprints/dotnet/src/APIGatewayAuthorizerHandler/Error/UnauthorizedException.cs [hunk 2]
```diff
@@ -0,0 +1,9 @@
+﻿namespace APIGatewayAuthorizerHandler.Error
+{
+    internal class UnauthorizedException : System.Exception
+    {
+        public UnauthorizedException() : base("Unauthorized")
```
Comment (jackrk) [inline blueprints/dotnet/src/APIGatewayAuthorizerHandler/Error/UnauthorizedException.cs]: Assuming you have tested that this results in a 401 response? API Gateway matches on the `errorMessage` provided by Lambda runtime.

**Comment by jackrk** [comment]
Closes #5

**Comment by jackrk** [comment]
Sorry @JoeDM09  can you please remove the .suo file, then I will merge.

File: blueprints/dotnet/src/APIGatewayAuthorizerHandler/Error/UnauthorizedException.cs (hunk 2 above)
Comment (JoeDM09) [reply blueprints/dotnet/src/APIGatewayAuthorizerHandler/Error/UnauthorizedException.cs]: @jackrk Yep, This is the reason for having a parameterless constructor which always passes "Unauthorized" as the exception message. Any other exception message won't trigger the expected 401.

**Comment by makcakaya** [comment]
This pull request works perfectly for me.

@JoeDM09 I have created a [repo](https://github.com/makcakaya/FirebaseAwsCustomAuth) using your pull request to use Firebase Auth as a source for AWS API Gateway custom authorization. I've given you the credits, hope you don't mind :) 

Thanks for the work!

**Comment by JoeDM09** [comment]
@jackrk @karthiksaligrama I notice this has been sitting as approved for some time so just wanted to give it a bit of a bump to see if we can get it merged now?

**Pull request merged**